crates/config/src/cli.rs

```diff
@@ -0,0 +1,315 @@
+#![deny(missing_docs)]
+//! Command-line interface utilities for configuration management.
+//!
+//! This module provides utility functions for handling configuration overrides
+//! via command-line arguments, supporting the `--set key=value` functionality
+//! across all Hypha binaries.
+
+use figment::providers::{Format, Toml};
+use miette::Result;
+use serde::{Deserialize, Serialize};
+
+use crate::LayeredConfigBuilder;
+
+/// Parse a single key-value pair for configuration overrides.
+///
+/// Parses command-line arguments in the format `KEY=value` into separate
+/// key and value strings for use with configuration override systems.
+///
+/// # Arguments
+///
+/// * `s` - A string slice in the format "key=value"
+///
+/// # Returns
+///
+/// Returns `Ok((key, value))` on successful parsing, or an error message
+/// describing what went wrong.
+///
+/// # Examples
+///
+/// ```rust
+/// use hypha_config::cli::parse_key_val;
+///
+/// let (key, value) = parse_key_val("listen_port=8080").unwrap();
+/// assert_eq!(key, "listen_port");
+/// assert_eq!(value, "8080");
+///
+/// let (key, value) = parse_key_val("debug=true").unwrap();
+/// assert_eq!(key, "debug");
+/// assert_eq!(value, "true");
+///
+/// // Handles nested configuration via dot notation
+/// let (key, value) = parse_key_val("listen_addresses.0=/ip4/127.0.0.1/tcp/8080").unwrap();
+/// assert_eq!(key, "listen_addresses.0");
+/// assert_eq!(value, "/ip4/127.0.0.1/tcp/8080");
+///
+/// // Returns error for invalid format
+/// assert!(parse_key_val("invalid_format").is_err());
+/// ```
+pub fn parse_key_val(s: &str) -> std::result::Result<(String, String), String> {
+    let pos = s
+        .find('=')
+        .ok_or_else(|| format!("invalid KEY=value: no `=` found in `{s}`"))?;
+    Ok((s[..pos].to_string(), s[pos + 1..].to_string()))
+}
+
+/// Convert key-value pairs to TOML format for configuration overrides.
+///
+/// Takes a collection of key-value pairs and converts them into a TOML
+/// string that can be used as a configuration provider. Performs basic
+/// type detection to avoid unnecessarily quoting numeric and boolean values.
+///
+/// # Arguments
+///
+/// * `overrides` - A slice of `(key, value)` tuples to convert
+///
+/// # Returns
+///
+/// Returns a TOML-formatted string ready for use with figment providers.
+///
+/// # Examples
+///
+/// ```rust
+/// use hypha_config::cli::create_overrides_toml;
+///
+/// let overrides = vec![
+///     ("debug".to_string(), "true".to_string()),
+///     ("port".to_string(), "8080".to_string()),
+///     ("name".to_string(), "my-service".to_string()),
+/// ];
+///
+/// let toml = create_overrides_toml(&overrides).unwrap();
+/// assert!(toml.contains("debug = true"));    // Boolean without quotes
+/// assert!(toml.contains("port = 8080"));     // Number without quotes  
+/// assert!(toml.contains("name = \"my-service\"")); // String with quotes
+/// ```
+pub fn create_overrides_toml(overrides: &[(String, String)]) -> Result<String> {
```

Comment:
This function doesn't have an error condition, therefore we can remove the `Result` here.

```suggestion
pub fn create_overrides_toml(overrides: &[(String, String)]) -> String {
```

Reply:
changed 👍